[FX] Changes in order to move python key out of tree

Chillee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56718 [WIP] Added skeleton of python dispatch key
* #57426 [nnc] Added NNC lowerings for t/transpose/permute/expand + other cleaning
* **#57427 [FX] Changes in order to move python key out of tree**
* #56875 [nnc] ported conv2d lowering over
* #56775 [NNC] ported over sum and softmax to new scheme

Differential Revision: [D28215322](https://our.internmc.facebook.com/intern/diff/D28215322)

Chillee committed:
[FX] Changes in order to move python key out of tree

[ghstack-poisoned]

Chillee committed:
Update on "[FX] Changes in order to move python key out of tree"




[ghstack-poisoned]

Chillee committed:
Update on "[FX] Changes in order to move python key out of tree"




[ghstack-poisoned]

Chillee committed:
Update on "[FX] Changes in order to move python key out of tree"




[ghstack-poisoned]

Chillee committed:
Update on "[FX] Changes in order to move python key out of tree"




[ghstack-poisoned]

suo commented on torch/fx/graph.py:
```diff
@@ -791,7 +792,7 @@ def add_global(name_hint: str, obj: Any):
             Returns: the global name that should be used to reference 'obj' in generated source.
             """
-            if _is_from_torch(obj):
+            if _is_from_torch(obj) and name_hint != 'device':  # to support registering torch.device
```
This check is a little suspicious—`name_hint` does not have any defined meaning, it's just a hint for how to serialize and may be mangled, autogenerated to be numbers, whatever.

can you do a direct object check instead?

Chillee replied:
ok :)

Chillee committed:
Update on "[FX] Changes in order to move python key out of tree"




[ghstack-poisoned]

Chillee committed:
Update on "[FX] Changes in order to move python key out of tree"




[ghstack-poisoned]

Chillee committed:
Update on "[FX] Changes in order to move python key out of tree"


Differential Revision: [D28215322](https://our.internmc.facebook.com/intern/diff/D28215322)

[ghstack-poisoned]

Chillee committed:
Update on "[FX] Changes in order to move python key out of tree"


Differential Revision: [D28215322](https://our.internmc.facebook.com/intern/diff/D28215322)

[ghstack-poisoned]

Chillee commented:
@chillee has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28215322).

Chillee committed:
Update on "[FX] Changes in order to move python key out of tree"


Differential Revision: [D28215322](https://our.internmc.facebook.com/intern/diff/D28215322)

[ghstack-poisoned]
